### DISCUSSION TITLE
[Fundamentos do C# - Aula sobre Var] Uso do var ou do tipo explicitamente

### DISCUSSION THREAD

**Discussion by lvsantos** [balta-io/forum#415] [Tira Dúvidas] [answered]
Olá Balta,

Na aula sobre o uso do var você cita sobre eliminar a redundância de informação na declaração de variáveis, como no exemplo a seguir: 

```
1. List<string> lista = new List<string>();
2. É mais aconselhável usar: var lista = new List<string>();
3. Ou então: List<string> lista = new();
```

Porém, já vi algumas discussões no Linkedin a respeito dessa redundância no retorno da chamada de métodos, especialmente em situações onde o tipo de retorno não é tão claro, por exemplo:

```
public void Exemplo()
{
    List<string> nomes = GerarNomes(5); // Exemplo 01
    var nomes = GerarNomes(5); // Exemplo 02
}

List<string> GerarNomes(int quantidade)
{
    //Lógica qualquer
}
```

Algumas pessoas defendem que o Exemplo 01 é redudante e que deveria ser usado o var para deixar o código mais limpo, porém outras pessoas defendem que o Exemplo 02 não deixa claro o retorno da função e que isso dificulta e legibilidade do código, já que nomes poderia ser uma List, IEnumerable, IQueryable, entre outros. Obviamente, nesse caso, pensando que ambos os métodos estivessem numa classe maior em que não fosse possível ver a declaração de GerarNomes sem ter que dar scroll.

Qual a sua opinião a respeito?

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @lvsantos , como vai?

Eu ainda prefiro o `var`. Mesmo no cenário 2, onde você não sabe o tipo do retorno, a IDE vai te ajudar, ela vai colocar o tipo do retorno ali na frente. O Rider e o VS fazem isto (Acredito que no VS Code tenha um plugin para isto também).

Mas de qualquer forma, não tem certo ou errado, qual você prefere?
